docs/data-sources/oauth_client.md

```diff
@@ -0,0 +1,330 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "pingfederate_oauth_client Data Source - terraform-provider-pingfederate"
+subcategory: ""
+description: |-
+  Describes an OAuth Client.
+---
+
+# pingfederate_oauth_client (Data Source)
+
+Describes an OAuth Client.
+
+## Example Usage
+
+```terraform
+data "pingfederate_oauth_client" "myOauthClientExample" {
+  client_id = "example"
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `client_id` (String) A unique identifier the client provides to the Resource Server to identify itself. This identifier is included with every request the client makes. For PUT requests, this field is optional and it will be overridden by the 'id' parameter of the PUT request.
+
+### Optional
```

Comment:
These should presumably all be read-only

Reply:
oops.. nice catch

Reply:
fixed